File: sellergoods-interface/src/main/java/com/jx/sellergoods/service/BrandService.java
package com.jx.sellergoods.service;

import com.jx.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

/**
 * Created by intellij IDEA
 *
 * @author Raven
 * Date:2018/12/27
 * Time:15:45
 * 品牌列表
 */
public interface BrandService {

    List<TbBrand> findAll();

    /**
     * 返回分页列表
     * @param pageNum 当前页面
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPages(int pageNum,int pageSize);

    /**
     * 新增品牌
     * @param brand
     */
    void add(TbBrand brand) throws Exception;

    /**
     * 修改品牌第一步 查找
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 修改品牌第二步 更新
     * @param brand
     */
    void update(TbBrand brand) throws Exception;

    /**
     * 删除
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 按条件查询
     * @param brand 可能的条件
     * @param pageNum 当前页面
     * @param pageSize 当前页大小
     * @return
     */
    PageResult findPages(TbBrand brand,int pageNum,int pageSize);
}

File: manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function($http){
    //读取列表数据绑定到表单中
    //查找所有
    this.findAll=function(){
        return $http.get('../brand/findAll.do');
    };

    //分页
    this.findPage = function (page,size) {
        return $http.get('../brand/findPage.do?page='+page+'&size='+size);
    };

    //查找实体
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    };

    //增加
    this.add = function (entity) {
        return $http.post('../brand/add.do',entity);
    };

    //更新
    this.update = function (entity) {
        return $http.post('../brand/update.do',entity);
    };

    //删除
    this.del = function (ids) {
        return  $http.get('../brand/delete.do?ids='+ids);
    };

    this.search = function (page,size,searchEntity) {
        return $http.post('../brand/search.do?page='+page+'&size='+size,searchEntity);
    }
});